### DISCUSSION TITLE
cannot set payload to my grid-Value from Solax-Inverter

### DISCUSSION THREAD

**Discussion by nothu71** [syssi/homeassistant-goecharger-mqtt#142] [General]
Hello,
I've just tried the code for the automations.yaml which is given as an example, but had no success.
My entity from the solax-inverter:  **sensor.solax_srmhc6vjq3_feed_in_power**.
I would have to invert the value (because the inverter gives negative values for grid-imports) and feed it to the payload.
For test-purpose I tried out
      **payload: '{"pGrid": 3000.0}'**
It worked, the charger tried to reduce his power-consumption to 3000 W (however he returned to 10kW because the pGrid-value of my testing-payload did not react as a result of his change.

The suggested line in the doku is
     **payload: {{'{"pGrid": '}}{{states('sensor.meter_active_power_raw')}}{{', "pPv":'}}{{states('sensor.total_dc_power')}}{{', "pAkku":0}'}}**

when I replace the entity and shorten the payload (all we need is pGrid) this would lead to
    **payload: {{'{"pGrid": '}}{{states('sensor.solax_srmhc6vjq3_feed_in_power')}}**

However this gives an Error  **bad indentation of a mapping entry** (24:32)

In the template section of homeAssistant I tried out
**{{'{"pGrid": '}}{{states('sensor.solax_srmhc6vjq3_feed_in_power') | multiply(-1)}}{{'}'}}**
This gives a perfect json-format:  **{"pGrid": 987.0}**

Unluckily it doesn't work when I copy this line to my payload-line in automations.yaml

Can someone help me??

**Comment by mihaimihai18** [comment]
Hi, use this one that is working for me, someone from a FB group helped me with it as I am no programmer (change your charger serial number and sensors names, I have a Huwaei invertor) https://l.facebook.com/l.php?u=https%3A%2F%2Fpastebin.com%2FvRmsjJ8X%3Ffbclid%3DIwZXh0bgNhZW0CMTAAAR3b9Y-h2nPbWKaPiiz-tnHVsrSCIhJ5QVISRWm5B-LL0oDbZZCNGGvc_tE_aem_AU8eydYWA3iafrFpyVfAO8A_g1n3oMBbi-qhXy2-egLWNg_Df1pGmD1Oqt8YcDzTIzpNM3wUWwCTllYAJK8_Bg5x&h=AT0vEBHsBDD2oOgKH9k1GzY82Q2PdYXiLJzQR90ne_qLXAx0x7taJbslTEbmfW1J3-kXcOmb03hJSqVDeY4w0SZg3Z4HP_oepv173P2FopteybbKOXyAO7Cvuu3TNTD4nQ&__tn__=R]-R&c[0]=AT3bDm6I9VSz5TyVSsRW8790Z-QbWVgGuldPMDe2TdG0Zzl5t0wjtSAai1YCo6mWinP25SR4BWavQyn-vNthY9QPyv-_1rIRdU52pWkBZcFmyKJoQYEApys9TrLGBcHofd0bRsPn_yNSDfsS2H77TXA15iIfmtx5kesf7qcGsOIdDYcpIdFhdDg8nQ

**Comment by Dengo91** [comment]
Hi, did you solved the problem? Because i need the same i have a invertor from Solaredge

**Comment by mihaimihai18** [comment]
Yes, as I said now it works ok with my Huawei inverter and go-eCharger.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________

**Reply by Dengo91** [reply]
Where do I have to implement this? Unfortunately, I don't have much experience with Home Assistant :) 

And you mean the code: 
alias: go-e Surplus Charging
description: "Simple automation to update values needed for using solar surplus with go-e Chargers"
trigger:
  - platform: time_pattern
    seconds: /5
condition: []
action:
  - service: mqtt.publish
    data:
      qos: "0"
      # Change to your charger ID here
      topic: go-eCharger/106455/ids/set
      # Please provide your own entities here, as described above
      payload: |
        {{'{"pGrid": '}}{{states('sensor.power_meter_active_power')}}{{', "pPv":'}}{{states('sensor.inverter_input_power')}}{{', "pAkku":0}'}}
      retain: false

**Comment by mihaimihai18** [comment]
The below code should be /homeassistant/automations.yaml file, make sure the indentation is correct:
  alias: ' go-e Surplus Charging'
  description: Simple automation to update values needed for using solar surplus with
    go-e Chargers
  trigger:
  - platform: time_pattern
    seconds: /10
  condition:
  - condition: sun
    before: sunset
    after: sunrise
  - type: is_on
    condition: device
    device_id: dbf7e9dc92384b89ff2195a917b0f4bf
    entity_id: 47e37f0df9d68f4e94a1c776191af880
    domain: binary_sensor
  action:
  - service: mqtt.publish
    data:
      qos: '0'
      topic: go-eCharger/xxxxxx/ids/set
      payload: '{{''{"pGrid": ''}}{{states(''sensor.power_meter_active_power'') |
        multiply(-1)}}{{'', "pPv":''}}{{states(''sensor.inverter_input_power'')}}{{'',
        "pAkku":0}''}} retain: false

I added „|multiply(-1)}} “ to change the energy available from solar from -xx to +xx (depends on how the inverter consider the injected energy sign). Also I added two conditions, the sun is up and the car is connected so it will works only if the above conditions are both met.
Good luck,
Mihai
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________

**Reply by Dengo91** [reply]
I tried it but i get an error:
Konfigurationsfehler
Error loading /config/configuration.yaml: while scanning a quoted scalar
  in "/config/automations.yaml", line 21, column 16
found unexpected end of stream
  in "/config/automations.yaml", line 22, column 20
  
![Bild 1](https://github.com/syssi/homeassistant-goecharger-mqtt/assets/171798362/0b6e6daa-bbaf-478a-ad33-d24a5bb10f80)

**Comment by mihaimihai18** [comment]
***@***.***

Check every space and enter, it works for me. Sorry I cannot help you more.
Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

________________________________
